File: primalTestDotNet/Entities/Exit.cs
using PrimalTestDotNet.Level;

namespace PrimalTestDotNet.Entities;

class Exit(ILeaveHandler container, int x, int y) : IGameObject
{
    public char Sprite { get => 'j'; }

    public IntVector2 Position { get; } = new(x, y);

    public void Draw(EntityRenderer renderer)
    {
        renderer.Draw(this);
    }

    public void Interact(Hero hero)
    {
        container.OnLeave(hero);
    }
}

File: PrimalTestCpp/Exit.cpp
#include "Exit.h"
#include "EntityRenderer.h"

#include "ILeaveHandler.h"

Exit::Exit(ILeaveHandler& container, int x, int y) : _container(container), _position(x, y) {}

char Exit::getSprite() const
{
    return 'j';
}

const IntVector2D& Exit::getPosition() const
{
    return _position;
}

void Exit::draw(const EntityRenderer& entityRenderer) const
{
    entityRenderer.draw(*this);
}

void Exit::interact(Hero& hero)
{
    _container.onLeave(hero);
}
